mslib/mscolab/mscolab.py

```diff
@@ -96,69 +96,236 @@ def handle_db_seed():
 
 def handle_mscolab_certificate_init():
     print('generating CRTs for the mscolab server......')
-    cmd = f"openssl req -newkey rsa:4096 -keyout {mscolab_settings.MSCOLAB_SSO_DIR}/key_mscolab.key -nodes -x509 -days 365 -batch -subj '/CN=localhost' -out {mscolab_settings.MSCOLAB_SSO_DIR}/crt_mscolab.crt"
-    os.system(cmd)
-    print('CRTs generated successfully for the mscolab server......')
+
+    try:
+        cmd = f"openssl req -newkey rsa:4096 -keyout {mscolab_settings.MSCOLAB_SSO_DIR}/key_mscolab.key -nodes -x509 -days 365 -batch -subj '/CN=localhost' -out {mscolab_settings.MSCOLAB_SSO_DIR}/crt_mscolab.crt"
+        subprocess.run(cmd, shell=True, check=True)
```

Comment:
I would still prefer getting rid of `shell=True`, which should be possible with:

```suggestion
        cmd = ["openssl", "req", "-newkey", "rsa:4096", "-keyout", f"{mscolab_settings.MSCOLAB_SSO_DIR}/key_mscolab.key", "-nodes", "-x509", "-days", "365", "-batch", "-subj", "/CN=localhost", "-out", f"{mscolab_settings.MSCOLAB_SSO_DIR}/crt_mscolab.crt"]
        subprocess.run(cmd, check=True)
```

Same for the other occurrences of `shell=True`. Only the `make_metadata` calls should be less straightforward due to the stdout redirection, which is a shell feature. That should also be doable in python using subprocess.run's `stdout=<file-object>` though.

Reply:
> 

This needs to be done after the exams too. We get back to this with writing tests.

Reply:
Resolved by #1974